spec/unit/pops/types/task_spec.rb

```diff
@@ -246,6 +246,25 @@ def compile(code = nil)
           end
         end
 
+        context 'that has no parameters' do
+          let(:testmodule) {
+            {
+              'tasks' => {
+                'hello' => 'echo hello',
+                'hello.json' => '{ "supports_noop": false }'             }
+            }
+          }
+
+          it 'loads the task with parameters set to undef' do
+            compile do
+              module_loader = Puppet.lookup(:loaders).find_loader('testmodule')
+              task = module_loader.load(:task, 'testmodule::hello')
+              expect(task_t.instance?(task)).to be_truthy
+              expect(task.parameters).to be_nil
+            end
+          end
```

Comment:
test missing for the semantics of having no parameters defined. Does that mean that "anything goes", or that no parameters are allowed? (I think it is "anything goes").

Reply:
Yes, anything goes when parameters are not defined. I'll add a test.

Reply:
Actually, the fact that no parameters means "anything goes" is implemented in the {{Puppet::Pal::TaskSignature}}, and there is a test for it. The task itself has no opinion about this.